core/src/main/java/com/yahoo/oak/Block.java

```diff
@@ -37,12 +37,21 @@ void setID(int id) {
     // Block manages its linear allocation. Thread safe.
     // The returned buffer doesn't have all zero bytes.
     boolean allocate(Slice s, int size) {
-        int now = allocated.getAndAdd(size);
-        if (now + size > this.capacity) {
-            allocated.getAndAdd(-size);
+        assert size >= 0; // avoid negative size
+        long next = 0;
+        boolean updated = false;
+
+        // update if capacity can accommodate requested size. use long to avoid integer overflow
+        while (!updated && (next = ((long) allocated.get()) + size) <= this.capacity) {
+            // compareAndSet manages value change as atomic transaction
+            updated = allocated.compareAndSet((int) (next - size), (int) next);
+        }
```

Comment:
Could you please modify this to use the `do {} while ()` syntax instead as it would be clearer?
Involving an assignment inside a while statement is prone to errors and considered hard to follow.

```java
do {
    cur = allocated.get();
    next = (long) cur + size;
} while (next <= capacity && !allocated.compareAndSet(cur, (int) next));
```

To validate the capacity later, you can simply use `if (next > capacity) throw ...`.

Reply:
@liran-funaro, Sure, your feedback to use `do while` will make the code more readable. Will update to use `do while`. Thank you.